src/algorithm/half_node.rs

```diff
@@ -110,8 +119,13 @@ mod test {
         // N(entry) -> head -> split            > N(merge) -> N(tail) -> N(exit)
         //               |          \-> right -/                 |
         //               \---<---<---<---<---<---<---<---<---<---/
-        // Allowing to identity two nested regions (and fixing the problem with a SimpleCfgView on the same example)
-        let v = HalfNodeView::new(&h);
+        // Allowing to identity two nested regions (and fixing the problem with an IdentityCfgMap on the same example)
+
+        // ALAN and another case for WholeHugrView
```

Comment:
Yes, I think it does not belong here. I thought that, since the RootHandle tag-checking had been added to main *since* this PR started, I could roll into this PR updates to the existing structure-detection code to use said tag-checking....but that leads to wanting #548 which has ended up significantly more complicated/longer-lived than expected, so I think it's best to leave this for #548 or another PR. I don't have an issue but I can roll it into that PR easily if this goes in first.